### ISSUE TITLE
failed calling webhook "validate.externalsecret.external-secrets.io": Post "https://external-secrets-webhook.default.svc:443/validate-external-secrets-io-v1beta1-externalsecret?timeout=5s"

### ISSUE THREAD

**Issue by Skullflow** [external-secrets/external-secrets#956]
Hi,

i'm running external-secrets helm chart 0.5.0 on an EKS 1.22 cluster, and i encountering this error when i create an ExternalService manifest:

```
failed calling webhook "validate.externalsecret.external-secrets.io": Post "https://external-secrets-webhook.default.svc:443/validate-external-secrets-io-v1beta1-externalsecret?timeout=5s"
```

This is my ExternalSecret:
```
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: my-secret
spec:
  refreshInterval: 1m
  secretStoreRef:
    name: secrets-manager-store
    kind: ClusterSecretStore
  target:
    name: my-secret
    creationPolicy: Owner
  data:
    - secretKey: key
      remoteRef:
        key: my-secret-provider-key
```
And this is my ClusterSecretStore
```
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  name: secrets-manager-store
spec:
  provider:
    aws:
      service: SecretsManager
      region: eu-central-1
      auth:
        secretRef:
          accessKeyIDSecretRef:
            name: my-aws-creds
            key: id
            namespace: default
          secretAccessKeySecretRef:
            name: my-aws-creds
            key: key
            namespace: default
```

Anyone can help?
Best.

**Comment by gusfcarvalho** [comment]
can you do a `kubectl get pods` for us? There should be a pod called `external-secrets-webhook`. Please can you fetch the logs for us?

**Comment by Skullflow** [comment]
Sure,

below the logs from `external-secrets` pod:
```
{"level":"info","ts":1649250021.7945418,"logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1649250021.795428,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1649250021.7958739,"msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8080"}
{"level":"info","ts":1649250021.7961748,"logger":"controller.secretstore","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"SecretStore","source":"kind source: *v1beta1.SecretStore"}
{"level":"info","ts":1649250021.7962956,"logger":"controller.secretstore","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"SecretStore"}
{"level":"info","ts":1649250021.7966506,"logger":"controller.externalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","source":"kind source: *v1beta1.ExternalSecret"}
{"level":"info","ts":1649250021.796764,"logger":"controller.externalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","source":"kind source: *v1.PartialObjectMetadata"}
{"level":"info","ts":1649250021.7968493,"logger":"controller.externalsecret","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret"}
{"level":"info","ts":1649250021.7973826,"logger":"controller.clustersecretstore","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ClusterSecretStore","source":"kind source: *v1beta1.ClusterSecretStore"}
{"level":"info","ts":1649250021.7975075,"logger":"controller.clustersecretstore","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"ClusterSecretStore"}
{"level":"info","ts":1649250021.7983546,"logger":"controller.clusterexternalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ClusterExternalSecret","source":"kind source: *v1beta1.ClusterExternalSecret"}
{"level":"info","ts":1649250021.7985091,"logger":"controller.clusterexternalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ClusterExternalSecret","source":"kind source: *v1.PartialObjectMetadata"}
{"level":"info","ts":1649250021.7985945,"logger":"controller.clusterexternalsecret","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"ClusterExternalSecret"}
{"level":"info","ts":1649250021.8980458,"logger":"controller.secretstore","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"SecretStore","worker count":1}
{"level":"info","ts":1649250021.898273,"logger":"controller.clustersecretstore","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"ClusterSecretStore","worker count":1}
{"level":"info","ts":1649250021.8987973,"logger":"controller.externalsecret","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","worker count":1}
{"level":"info","ts":1649250021.902173,"logger":"controller.clusterexternalsecret","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"ClusterExternalSecret","worker count":1}
{"level":"info","ts":1649250021.9318333,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649250021.9627717,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649250024.8149598,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649250321.991122,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649250622.032611,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649250922.0622764,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649251222.0900989,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649251522.131159,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
{"level":"info","ts":1649251822.1665611,"logger":"provider.aws","msg":"using aws session","region":"eu-central-1","credentials":{}}
```

And here the logs of `external-secrets-webhook` pod:
```
{"level":"info","ts":1649250021.1875336,"logger":"setup","msg":"validating certs"}
{"level":"error","ts":1649250021.1875982,"logger":"setup","msg":"invalid certs. retrying...","error":"stat /tmp/certs/tls.crt: no such file or directory","stacktrace":"github.com/external-secrets/external-secrets/cmd.waitForCerts\n\t/home/runner/work/external-secrets/external-secrets/cmd/webhook.go:161\ngithub.com/external-secrets/external-secrets/cmd.glob..func3\n\t/home/runner/work/external-secrets/external-secrets/cmd/webhook.go:69\ngithub.com/spf13/cobra.(*Command).execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974\ngithub.com/spf13/cobra.(*Command).Execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902\ngithub.com/external-secrets/external-secrets/cmd.Execute\n\t/home/runner/work/external-secrets/external-secrets/cmd/root.go:169\nmain.main\n\t/home/runner/work/external-secrets/external-secrets/main.go:21\nruntime.main\n\t/opt/hostedtoolcache/go/1.17.8/x64/src/runtime/proc.go:255"}
{"level":"info","ts":1649250031.1880028,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649250031.591933,"logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1649250031.5923476,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1beta1, Kind=ExternalSecret"}
{"level":"info","ts":1649250031.5923913,"logger":"controller-runtime.builder","msg":"Registering a validating webhook","GVK":"external-secrets.io/v1beta1, Kind=ExternalSecret","path":"/validate-external-secrets-io-v1beta1-externalsecret"}
{"level":"info","ts":1649250031.5925176,"logger":"controller-runtime.webhook","msg":"Registering webhook","path":"/validate-external-secrets-io-v1beta1-externalsecret"}
{"level":"info","ts":1649250031.5927022,"logger":"controller-runtime.webhook","msg":"Registering webhook","path":"/convert"}
{"level":"info","ts":1649250031.5927722,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1beta1, Kind=ExternalSecret"}
{"level":"info","ts":1649250031.592801,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1beta1, Kind=SecretStore"}
{"level":"info","ts":1649250031.5928118,"logger":"controller-runtime.builder","msg":"Registering a validating webhook","GVK":"external-secrets.io/v1beta1, Kind=SecretStore","path":"/validate-external-secrets-io-v1beta1-secretstore"}
{"level":"info","ts":1649250031.5928724,"logger":"controller-runtime.webhook","msg":"Registering webhook","path":"/validate-external-secrets-io-v1beta1-secretstore"}
{"level":"info","ts":1649250031.5929687,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1beta1, Kind=SecretStore"}
{"level":"info","ts":1649250031.5929844,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1beta1, Kind=ClusterSecretStore"}
{"level":"info","ts":1649250031.5929935,"logger":"controller-runtime.builder","msg":"Registering a validating webhook","GVK":"external-secrets.io/v1beta1, Kind=ClusterSecretStore","path":"/validate-external-secrets-io-v1beta1-clustersecretstore"}
{"level":"info","ts":1649250031.5930467,"logger":"controller-runtime.webhook","msg":"Registering webhook","path":"/validate-external-secrets-io-v1beta1-clustersecretstore"}
{"level":"info","ts":1649250031.5931487,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1beta1, Kind=ClusterSecretStore"}
{"level":"info","ts":1649250031.593164,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=ExternalSecret"}
{"level":"info","ts":1649250031.5931795,"logger":"controller-runtime.builder","msg":"skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=ExternalSecret"}
{"level":"info","ts":1649250031.5932171,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1alpha1, Kind=ExternalSecret"}
{"level":"info","ts":1649250031.5932267,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=SecretStore"}
{"level":"info","ts":1649250031.593238,"logger":"controller-runtime.builder","msg":"skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=SecretStore"}
{"level":"info","ts":1649250031.593266,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1alpha1, Kind=SecretStore"}
{"level":"info","ts":1649250031.5932753,"logger":"controller-runtime.builder","msg":"skip registering a mutating webhook, object does not implement admission.Defaulter or WithDefaulter wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=ClusterSecretStore"}
{"level":"info","ts":1649250031.5932832,"logger":"controller-runtime.builder","msg":"skip registering a validating webhook, object does not implement admission.Validator or WithValidator wasn't called","GVK":"external-secrets.io/v1alpha1, Kind=ClusterSecretStore"}
{"level":"info","ts":1649250031.5933092,"logger":"controller-runtime.builder","msg":"Conversion webhook enabled","GVK":"external-secrets.io/v1alpha1, Kind=ClusterSecretStore"}
{"level":"info","ts":1649250031.5933204,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1649250031.593416,"logger":"controller-runtime.webhook.webhooks","msg":"Starting webhook server"}
{"level":"info","ts":1649250031.5934854,"msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"info","ts":1649250031.5935106,"msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8080"}
{"level":"info","ts":1649250031.5937872,"logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
{"level":"info","ts":1649250031.5939925,"logger":"controller-runtime.webhook","msg":"Serving webhook server","host":"","port":9443}
{"level":"info","ts":1649250031.594184,"logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
{"level":"info","ts":1649250331.189845,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649250331.1903253,"logger":"setup","msg":"certs are valid"}
{"level":"info","ts":1649250631.189173,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649250631.1897285,"logger":"setup","msg":"certs are valid"}
{"level":"info","ts":1649250931.1899374,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649250931.1904373,"logger":"setup","msg":"certs are valid"}
{"level":"info","ts":1649251231.1899915,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649251231.19059,"logger":"setup","msg":"certs are valid"}
{"level":"info","ts":1649251531.1899655,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649251531.190372,"logger":"setup","msg":"certs are valid"}
{"level":"info","ts":1649251831.190177,"logger":"setup","msg":"validating certs"}
{"level":"info","ts":1649251831.191616,"logger":"setup","msg":"certs are valid"}
```

These pods are in the default namespace and i'm trying to create an external secrets in another namespace (using ClusterSecretStore).

**Comment by Skullflow** [comment]
Yes, i'm trying to create a SecretStore in the same namespace of the ExternalSecrets but without success.

> you might need to add some firewall permissions so EKS kube-apiserver can reach out webhook pod.

How can I do that?

**Comment by Skullflow** [comment]
I'm using ArgoCD to sync my manifest in the cluster, but when i run: `kubectl apply -f external-secrets.yaml`, this is the error:
```
Error from server (InternalError): error when creating "external-secrets.yaml": Internal error occurred: failed calling webhook "validate.externalsecret.external-secrets.io": Post "https://external-secrets-webhook.default.svc:443/validate-external-secrets-io-v1beta1-externalsecret?timeout=5s": context deadline exceeded
```

The content of `external-secrets.yaml`:
```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: my-secret
spec:
  refreshInterval: 1m
  secretStoreRef:
    name: secrets-manager-store
    kind: ClusterSecretStore
  target:
    name: my-secret
    creationPolicy: Owner
  data:
    - secretKey: key
      remoteRef:
        key: my-provider-key-secret
```

Furthermore, I want to report that on the docs in the overview section, this is the manifest example of `SecretStore`:
```yaml
apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: secretstore-sample
spec:
  provider:
    aws:
      service: SecretsManager
      region: us-east-1
      auth:
        secretRef:
          accessKeyIDSecretRef:
            name: awssm-secret
            key: access-key
          secretAccessKeySecretRef:
            name: awssm-secret
            key: secret-access-key
````
Instead, on the API Types > SecretStore section, the `auth.secretRef.accessKeyIDSecretRef` is `auth.secretRef.accessKeyID` and the `auth.secretRef. secretAccessKeySecretRef` is `auth.secretRef. secretAccessKey`. 

Which one should we use?

**Comment by gusfcarvalho** [comment]
@pnowy please see this link https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules. I suppose this is the same as for EKS, where firewall rules are needed to allow communication from the API server to the webhook.

**Comment by gusfcarvalho** [comment]
@Skullflow your error is a little bit different than the other cases we had. Can you confirm if you are using a public EKS or a private EKS instance?

**Comment by winelivery** [comment]
I created the cluster using this terraform module: https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest

Both of the `cluster_endpoint_private_access` and `cluster_endpoint_public_access` are `true`

So i assume that the k8s api server is reachable publicly

**Comment by Skullflow** [comment]
the node group is in a private network (VPC) but the API server is reachable publicly.

I use this terraform module to create the cluster: https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest

and both `cluster_endpoint_private_access` and `cluster_endpoint_public_access` as set to `true``

**Comment by gusfcarvalho** [comment]
If you are using Calico as a CNI, @Skullflow this link might be relevant to you: https://medium.com/@denisstortisilva/kubernetes-eks-calico-and-custom-admission-webhooks-a2956b49bd0d

**Comment by gusfcarvalho** [comment]
The symptom you are having is fairly similar to this issue: https://github.com/open-policy-agent/gatekeeper/issues/1091

**Comment by cooperbenson-qz** [comment]
I also faced this issue on my private GKE cluster, but was able to resolve the issue by adding a firewall rule allowing the master network to access port 9443 (the port the webhook pod is listening on). This may solve the issue for others, and if so might just require a docs change to call out the need to allow access to this port.

**Comment by Skullflow** [comment]
> What is your VPC CNI version?

the version is: v1.10.1-eksbuild.1

**Comment by Skullflow** [comment]
@gusfcarvalho how can install with `webhook.create=false` and `certController.create=false`? I can't find these helm values on the documentation. Could you please post an example?

**Comment by Skullflow** [comment]
Ok, I found the `values.yaml` file in this repo: https://github.com/external-secrets/external-secrets/blob/b3f7b7ac947ab171852202fe684e4374cd2c81c6/deploy/charts/external-secrets/values.yaml

_(i suggest to link this file or report the available attributes in the docs, it's always a pain to find out the helm customization options :D)_

Then, I added these lines to my `values.yaml`:
```yaml
certController:
  create: false
webhook:
  create: false
```

And now it works smoothly.

**Comment by djboboch** [comment]
Hi this is not a bug. I had the same problem the other day.

The problem is happening due to the private eks cluster on gcp. When you create a private cluster the control plane is located in a different subnet then the nodes with the kublet.
When you are trying to create a secret a webhook event is send however because by default the traffic between the control place and the kublet on the node is forbidden you are getting the error.

**SOLUTION**

Add a firewall rule that allow traffic between the master control plane CIDR block and the subnet where your resources are located.

_reference_: https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules

**Comment by pnowy** [comment]
Yep, forgot about private GKE firewall. Just example terraform setup for it if somebody needs that:

```
resource "google_compute_firewall" "cluster_firewall_rule" {
  name    = "foo"
  network = var.network
  allow {
    protocol = "tcp"
    ports    = ["9443"]
  }
  source_ranges = [var.master_ipv4_cidr_block]
}
```

That solved the issue.

**Comment by Skullflow** [comment]
Yeah, i confirm that.

Adding an aws security group rule (allow all ingress traffic between control plane and the node groups) solves the issue.

This is not a bug.

**Comment by moolen** [comment]
Glad to hear that! I think we should add a  troubleshooting guide / FAQ to the docs. The information in this issue would be a good candidate to go there :)

**Comment by ZeroDeth** [comment]
@Skullflow 

Can you please share the AWS security group because still getting the same error? 

`Error from server (InternalError): error when creating ".": Internal error occurred: failed calling webhook "validate.externalsecret.external-secrets.io": Post "https://external-secrets-1649355652-webhook.external-secrets.svc:443/validate-external-secrets-io-v1beta1-externalsecret?timeout=5s": context deadline exceeded
Error from server (InternalError): error when creating ".": Internal error occurred: failed calling webhook "validate.secretstore.external-secrets.io": Post "https://external-secrets-1649355652-webhook.external-secrets.svc:443/validate-external-secrets-io-v1beta1-secretstore?timeout=5s": context deadline exceeded
`

I am using terraform-aws-eks/vpc. 

Thanks

**Comment by Skullflow** [comment]
@ZeroDeth which version of terraform aws eks module are you using?

**Comment by Ravelin** [comment]
> allow all ingress traffic between control plane and the node groups

I'm having same issue, tried adding all different ways and still won't validate again control plane.  I am using 
`  source = "terraform-aws-modules/eks/aws"
  version = "18.17.0"`

Any help would be greatly appreciated

**Comment by Skullflow** [comment]
@ZeroDeth @Ravelin 

The EKS cluster created with `terraform-aws-modules` v18+ got some hidden security group rules between the control plane and the node groups.

In order to overwrite these rules you need to specify the rule (or rules) like this (it's quite loose):

```tf
 node_security_group_additional_rules = {
    ingress_cluster_all = {
      description                   = "Cluster to node all ports/protocols"
      protocol                      = "-1"
      from_port                     = 0
      to_port                       = 0
      type                          = "ingress"
      source_cluster_security_group = true
    }
}
```

The `source_cluster_security_group` attributes indicates that the source of the communication is the control plane security group.

**Comment by Ravelin** [comment]
@Skullflow 

Thanks, I worked it out about 20 minutes after I posted but now i'm not getting my secrets to appear in my namespace.  I think I need to adjust something.  I approached it slightly differently and tagged those into my `eks` using `vpc_security_group_ids = [aws_security_group.additional.id]` since I had some other rules setup there also

```
resource "aws_security_group" "additional" {
  name_prefix = "${local.name}-additional"
  vpc_id      = module.vpc.vpc_id

  ingress {
    from_port = 9443
    to_port   = 9443
    protocol  = "tcp"
    cidr_blocks      = ["0.0.0.0/0"]
    ipv6_cidr_blocks = ["::/0"]
  }
}
```

Thank you for the help and the other issue, gives me some reading to do and work out which is better

**Comment by ZeroDeth** [comment]
@Skullflow Thanks, and using Terraform v18.17.0. It was sorted with an example provided of the security group and working now.

**Comment by gusfcarvalho** [comment]
This issue is closed now. We introduced in #965 the possibility of configuring the port of the webhook, plus the defaults to listen to 10250 (which is already configured in terraform-eks provider).

Feel free to reopen it, though!

**Comment by LyKos4** [comment]
> * `hostNetwork=true`
How to change this in already deployed external-secrets? Shall I edit the secrets-webhook deployment ?

**Comment by maven2016** [comment]
> Ok, I found the `values.yaml` file in this repo: https://github.com/external-secrets/external-secrets/blob/b3f7b7ac947ab171852202fe684e4374cd2c81c6/deploy/charts/external-secrets/values.yaml
> 
> _(i suggest to link this file or report the available attributes in the docs, it's always a pain to find out the helm customization options :D)_
> 
> Then, I added these lines to my `values.yaml`:
> 
> ```yaml
> certController:
>   create: false
> webhook:
>   create: false
> ```
> 
> And now it works smoothly.

tried, i still get the error:
Internal error occurred: failed calling webhook "validate.secretstore.external-secrets.io": failed to call webhook: Post "https://external-secrets-webhook.default.svc:443/validate-external-secrets-io-v1beta1-secretstore?timeout=5s"
 @gusfcarvalho 
 how to disable the hook in version v0.9.20?

**Comment by Maxwell2022** [comment]
Using the latest version of this chart `v0.10.3`, we still had issue with Fargate. 
The default port for the webhook was `10250` which as mentioned above is a reserved port when using EKS Fargate. Also, we are using terraform to apply this helm and had to update our configuration to:

```hcl
resource "helm_release" "this" {
  force_update = true
  name         = "external-secrets"
  repository   = "https://charts.external-secrets.io"
  chart        = "external-secrets"
  version      = var.chart_version

  namespace        = "external-secrets"
  create_namespace = true
  atomic           = true // purges chart on fail
  timeout          = 300

  set {
    name  = "installCRDs"
    value = true
  }

  # default port (10250) is reserved for Fargate
  set {
    name  = "webhook.port"
    value = 9443
  }
}

```

Hope that helps someone else

**Comment by AzySir** [comment]
> Using the latest version of this chart `v0.10.3`, we still had issue with Fargate. The default port for the webhook was `10250` which as mentioned above is a reserved port when using EKS Fargate. Also, we are using terraform to apply this helm and had to update our configuration to:
> 
> ```hcl
> resource "helm_release" "this" {
>   force_update = true
>   name         = "external-secrets"
>   repository   = "https://charts.external-secrets.io"
>   chart        = "external-secrets"
>   version      = var.chart_version
> 
>   namespace        = "external-secrets"
>   create_namespace = true
>   atomic           = true // purges chart on fail
>   timeout          = 300
> 
>   set {
>     name  = "installCRDs"
>     value = true
>   }
> 
>   # default port (10250) is reserved for Fargate
>   set {
>     name  = "webhook.port"
>     value = 9443
>   }
> }
> ```
> 
> Hope that helps someone else

This has worked for me thank you for sharing!

I had to create the resources. Helm Version is `v0.10.7`

**Comment by lgo** [comment]
I recently stumbled on this, but couldn't quite figure out what went wrong because I'm using a self-hosted k3s cluster. Additionally, once I tried an earlier suggestion (`webhook: {enabled: false}`), I was able to create a `SecretStore` but then subsequent `ExternalSecret`'s failed.

I eventually realized the error indicated "502 (Bad Gatway)" for an internal IP address that pointed to the `external-secrets` deployment pod rather than the `external-secrets-webhook` pod... it was that point that I discovered this was my own mistake, so sharing this in case anyone else is as foolish as me.

I had a Kustomization setup like so
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

helmCharts:
- name: external-secrets
  releaseName: external-secrets
  version: 0.12.1
  namespace: external-secrets
  repo: https://charts.external-secrets.io
  includeCRDs: true

labels:
  - includeSelectors: true
    pairs:
      app.kubernetes.io/instance: external-secrets
      app.kubernetes.io/name: external-secrets
```

those last few lines included those labels on all deployment/pods, and _also_ inadverdently changes the selectors on the service. (Massive 🤦, I had been using this broadly on the few helm charts I had without realizing the potential consequences)
